server/controllers/conversation.js

```diff
@@ -2,25 +2,33 @@ const Conversation = require("../models/Conversation");
 const asyncHandler = require("express-async-handler");
 
 
-// Given two userIDs, create a conversation
+// Given two unique userIDs, create a private conversation
 exports.createConversation = asyncHandler(async (req, res) => {
-    const { senderID, receiverID } = req.body;
+    const senderID = req.user.id;
+    const { receiverID } = req.body;
+
+    if(senderID === receiverID) {
```

Comment:
We need to check whether a conversation between the two users already exists. Only if it doesn't exist, we need to create a new conversation

Reply:
You can remove this if/else construct as we are going to be sending unique sender and receiver id's from the FE